api/openapi.yaml

```diff
@@ -1277,6 +1277,21 @@ components:
           readOnly: true
       description: An object that is returned in cases of success that returns nothing.
 
+    BaseAPIObject:
```

Comment:
Why not just calling it Metadata?
Don't you think this will be part of the Metadata of the object?

Reply:
@fishkerez I will propose another object called `Metadata` which would be common field for all the root level API objects like the `Annotations`. The `BaseAPIObject` is used to combine all the mandatory/common fields into a single object definition which then can be used for composing new root level API objects and to extend the existing ones. I agree the `BaseAPIObject` is not the best name and I am open for any suggestions other then `Metadata`.

Reply:
@chrisgacsal I am struggling to understand the difference between Metadata and BaseAPIObject.
**Metadata** - would be common field for all the root level API objects like the Annotations.
**BaseAPIObject** - is used to combine all the mandatory/common fields into a single object definition which then can be used for composing new root level API objects and to extend the existing ones.

So which fields are belong to Metadata and which to the Base? To me it sounds like annotations are a common field that will be used by all object, so not sure to where it belongs.

BTW, in Kubernetes the annotations are part of the metadata.

Reply:
correct me if I'm wrong but I would expect it to look something like:

```
BaseAPIObject:
   type: object
   properties:
      metadata:
         type: object
         properties:
            createdAt:
               type: string
               format: date-time
            updatedAt:
            ...
            ...
      annotations:
         type: array
         ...
      id:
        type: string
```

whether annotations is part of metadata or not I'm not bothered, then its used like:

```
Asset:
  allOf:
     - $ref: "..../BaseAPIObject"
     - type: object
        properties:
           assetInfo:
              $ref: ...      
```

Reply:
Ahh @Tehsmash has already described what I did in the previous comment.
I still think it would be better to separate the `Annotations` from `Metadata` as the actor operating on these fields are different. Former is used/managed by components other than `API Server` while the latter must be only managed by the `API Server` only. Defining permissions for fields is easier if these two fields are not embedded in a high level field.

Reply:
@Tehsmash  @fishkerez @paralta Wdyt?

Reply:
Honestly I'm not convinced that we need a Metadata wrapper key around any of them if we go with the composite object approach. ID, createdAt, updatedAt, annotations, revision, can all be top level keys for all objects without a need for another layer.

We can create an interface and Get/Set functions for each object if we want to be able to process them in a common way in the go code.

Reply:
Personally, I think would be much cleaner to have a top level key (`metadata`) which includes the metadata properties like (`createdAt`, updatedAt, `revision`, etc). On the other hand the `annotations` definitely needs to be a top level field.

Reply:
Alright. Let's keep it as is for now.